### PR TITLE
Feature/navbar

### PR DESCRIPTION
**Pull request by jackcasstlesjones**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

- Added  Navbar, Navbar buttons, and Icon component
- Fixed tailwind config
- Added navbar to Root Layout
- Lots of other changes because of Prettier formatting - apologies for this
- Added page.tsx files for needs and insights

## Screenshots

Screenshot 2024-11-27 at 14 14 55

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_

- [ ] Semantic HTML implemented?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

## Added/updated tests?

_Please aim to keep the code coverage percentage at 80% and above._

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: src/ui/shared/Icon.tsx
Comment (maxitect) [inline src/ui/shared/Icon.tsx:1]: Do we need this component, what does it provide that the out of the box Icon component does not?

File: src/ui/layout/Navbar/NavbarButtons.tsx
Comment (maxitect) [inline src/ui/layout/Navbar/NavbarButtons.tsx:1]: Should the navbar navigation logic not live inside Navbar component and this should just be a component representing 1 button? Tbd maybe just a thought

**Review by maxitect — CHANGES_REQUESTED** [review]
Couple of comments to either change or discuss

File: src/ui/shared/Icon.tsx
Comment (jackcasstlesjones) [reply src/ui/shared/Icon.tsx:1]: done!

**Review by maxitect — APPROVED** [review]

**Pull request merged**